### DISCUSSION TITLE
"Troubles Building PyAnsys Geometry Docker Container: Seeking Help for Ansys 2024 R1 Student Version"

### DISCUSSION THREAD

**Discussion by santoshdoddipatla** [ansys/pyansys-geometry#978] [General]
I am a student currently working on integrating PyFluent with PyAnsys Geometry. I have successfully installed the Ansys 2024 R1 student version, GitHub Desktop, and Docker Desktop. Following the documentation, I cloned the PyAnsys Geometry repository into a local folder. 

However, when attempting to build the Windows Docker container using the command "docker build -t ghcr.io/ansys/geometry:windows-latest -f Dockerfile.windows .", I encountered an issue where the Windows binaries.zip file could not be downloaded. Unfortunately, the "latest release artifacts for the Windows Docker container" hyperlink in the documentation led to a 404 error. 

I am seeking guidance on whether it is possible to build, connect, and launch Ansys Geometry services using the Ansys 2024 R1 student version. If feasible, please help  with the installation process.

![Screenshot 2024-02-07 163517](https://github.com/ansys/pyansys-geometry/assets/143094710/d188463e-f4f0-4a9f-bd04-a86777a944e7)

![Screenshot 2024-02-07 160625](https://github.com/ansys/pyansys-geometry/assets/143094710/af39af77-1b21-4a40-ba6a-3f4cb4d7bb1e)

![Screenshot 2024-02-07 155821](https://github.com/ansys/pyansys-geometry/assets/143094710/9f892df8-4cce-4fe3-8c22-5b72030a08ac)

**Comment by RobPasMue** [comment]
Hi @santoshdoddipatla - let's go step by step on your comments:

> Unfortunately, the "latest release artifacts for the Windows Docker container" hyperlink in the documentation led to a 404 error.

As stated in the documentation, direct access to the binaries is only available for Ansys employees. I'm sorry for the inconveniences. If you had the Ansys Geometry Service locally installed, you could make use of the ``build_docker_windows.py`` script located here: https://github.com/ansys/pyansys-geometry/blob/main/docker/build_docker_windows.py

> I am seeking guidance on whether it is possible to build, connect, and launch Ansys Geometry services using the Ansys 2024 R1 student version. If feasible, please help with the installation process.

I verified the Ansys Student 24R1 installation and the Ansys Geometry Service is currently not distributed with it. What you do have access to though is both SpaceClaim and Ansys Discovery. You can also run SpaceClaim and Discovery in headless modes if you prefer.

Using ``ansys-geometry-core`` version 0.4.9 or later, you can perform the following:

```py

from ansys.geometry.core import launch_modeler

# For SpaceClaim
modeler = launch_modeler(mode="spaceclaim", hidden=True)

# For Ansys Discovery
modeler = launch_modeler(mode="discovery", hidden=True)
```

**Reply by santoshdoddipatla** [reply]
thank you for your quick response now i am able to draw geometry in  spaceclaim ,but facing problem on exporing the geometry to local machine thank you.......

**Reply by RobPasMue** [reply]
Hi @santoshdoddipatla! What kind of issue are you facing?

**Reply by RobPasMue** [reply]
The following script should allow you to download files:

```py
    from pathlib import Path
    from ansys.geometry.core.designer import DesignFileFormat

    file_dir = Path(os.getcwd(), "downloads")
    file_dir.mkdir(parents=True, exist_ok=True)
    design.download(file_location=Path(file_dir, "file.scdocx"), format=DesignFileFormat.SCDOCX)
 ```

**Reply by RobPasMue** [reply]
Also, try defining the following env var on your system:

Key ``SPACECLAIM_MODE``
Value ``2``

This will avoid showing a popup screen to specify file settings when running in hidden mode :)

**Reply by RobPasMue** [reply]
The following script shows how to run SpaceClaim on hidden mode, create a design and download/export the file:

```py
import os
from pathlib import Path

from ansys.geometry.core import launch_modeler
from ansys.geometry.core.math import Point2D
from ansys.geometry.core.sketch import Sketch
from ansys.geometry.core.designer import DesignFileFormat

# Launch the modeler
modeler = launch_modeler(mode="spaceclaim", hidden=True)

# Create a simple design
design = modeler.create_design("Test")
comp1 = design.add_component("Comp1")
comp3 = design.add_component("Comp3")
body1 = comp1.extrude_sketch("Body1", Sketch().box(Point2D([0, 0]), 1, 1), 1)
body3 = comp3.extrude_sketch("Body3", Sketch().box(Point2D([5, 0]), 1, 1), 1)

# Download the model
file_dir = Path(os.getcwd(), "downloads")
file_dir.mkdir(parents=True, exist_ok=True)
design.download(file_location=Path(file_dir, "file.scdocx"), format=DesignFileFormat.SCDOCX)
```

No popup window should appear as long the environment variable is defined in the system.

```console
(.venv) PS D:\repos\pyansys-geometry> $env:SPACECLAIM_MODE
2       
(.venv) PS D:\repos\pyansys-geometry> python .\test.py    
```
I tried this out myself and it worked perfectly fine. Give it a try @santoshdoddipatla =)

**Reply by RobPasMue** [reply]
We have implemented a fix in https://github.com/ansys/pyansys-geometry/pull/979 - that way whenever you run SpaceClaim in hidden mode we will ensure that the environment variables is set on the user's behalf.

**Reply by santoshdoddipatla** [reply]
its working fine sir thank you for your help

**Reply by santoshdoddipatla** [reply]
Creating a 3D Wing with Varying Chord Sections
i am working on project title of optimization of aerodynamic shape of fixed wing uav.Reference airfoil profile is naca 2412
i am facing problem in creating 3d wing geometry in pyansys geometry using spaceclaim modeler, firstly i started with 2d airfoil profile using pyvista spline command and then i tried to extrude it but i am getting error . is it possible to create a 3d wing in pyansys geometry with different chord sections. if it is possible can you help me how  to create a 3d wing

**Reply by RobPasMue** [reply]
Let's leave that for #984 if you don't mind to keep the discussions isolated @santoshdoddipatla

**Reply by RobPasMue** [reply]
https://geometry.docs.pyansys.com/version/dev/examples/04_applied/01_naca_airfoils.html

**Reply by threedots65** [reply]
I have the same problem, I use ansys student r1, but when I try to connect spaceclaim or discover I get an error
![изображение_2024-05-07_145239840](https://github.com/ansys/pyansys-geometry/assets/98290096/a9362ad6-0419-475e-ad00-1dd48e1c8b08)

**Reply by RobPasMue** [reply]
Hi @threedots65 - as stated above:

> I verified the Ansys Student 24R1 installation and the Ansys Geometry Service is currently not distributed with it. What you do have access to though is both SpaceClaim and Ansys Discovery. You can also run SpaceClaim and Discovery in headless modes if you prefer.

```py
from ansys.geometry.core import launch_modeler

# For SpaceClaim
modeler = launch_modeler(mode="spaceclaim", hidden=True)

# For Ansys Discovery
modeler = launch_modeler(mode="discovery", hidden=True)
```

Feel free to open a new discussion with your specific issue
